samples/performance/image_compression_control/image_compression_control.h

```diff
@@ -0,0 +1,189 @@
+	 * This flag can also be used to disable all compression, which is not recommended.
+	 */
+	VkImageCompressionFlagsEXT compression_flag{VK_IMAGE_COMPRESSION_DEFAULT_EXT};
+
+	/**
+	 * @brief Possible compression schemes
+	 */
+	enum class TargetCompression : int
+	{
+		Default   = 0,
+		FixedRate = 1,
+		None      = 2,
+
+		// Enum value count
+		Count = 3
+	};
+
+	/**
+	 * @brief Compression rate for the color image
+	 * If fixed-rate compression is selected, this specifies the bitrate per component.
+	 */
+	VkImageCompressionFixedRateFlagsEXT compression_fixed_rate_flag_color{};
+
+	/**
+	 * @brief Compression rate for the Swapchain image
+	 */
+	VkImageCompressionFixedRateFlagsEXT compression_fixed_rate_flag_swapchain{};
+
+	/**
+	 * @brief Supported values for compression rate
+	 * These depend on the properties (format and usage) of the color image.
+	 */
+	std::vector<VkImageCompressionFixedRateFlagBitsEXT> supported_fixed_rate_flags_color{};
+
+	/**
+	 * @brief Supported values for compression rate
+	 * These depend on the properties (format and usage) of the Swapchain image.
+	 */
+	std::vector<VkImageCompressionFixedRateFlagBitsEXT> supported_fixed_rate_flags_swapchain{};
+
+	/**
+	 * @brief Possible fixed-rate compression levels
+	 */
+	enum FixedRateCompressionLevel : int
+	{
+		High = 0,        // Highest compression available (smallest bitrate)
+		Low  = 1,        // Lowest compression available (largest bitrate)
+	};
+
+	/**
+	 * @brief Selects the compression bitrate given a list of supported values and a compression level
+	 */
+	VkImageCompressionFixedRateFlagBitsEXT select_fixed_rate_compression_flag(std::vector<VkImageCompressionFixedRateFlagBitsEXT> &supported_fixed_rate_flags,
+	                                                                          FixedRateCompressionLevel                            compression_level);
+
+	/**
+	 * @brief Size (in MB) of the color image (output of the forward rendering pass)
+	 */
+	float footprint_color{};
+
+	/**
+	 * @brief Size (in MB) of the Swapchain image (output of the postprocessing pass)
+	 */
+	float footprint_swapchain{};
+
+	/**
+	 * @brief Time
+	 * Used to animate the chromatic aberration effect.
+	 */
+	float elapsed_time{};
+
+	/**
+	 * @brief Updates compression settings based on user GUI input.
+	 *        Triggers recreation of the Swapchain and other render targets.
+	 */
+	void update_render_targets();
+
+	/**
+	 * @brief Unsupported compression schemes
+	 * Used to reduce the GUI controls if, for instance, the required
+	 * extensions are not supported by the device.
+	 */
+	std::set<TargetCompression> gui_skip_compression_values;
+
+	/* Helpers for managing GUI input */
+
+	int gui_target_compression{(int) TargetCompression::Default};
+
+	int last_gui_target_compression{gui_target_compression};
+
+	int gui_fixed_rate_compression_level{(int) FixedRateCompressionLevel::High};
+
+	int last_gui_fixed_rate_compression_level{gui_fixed_rate_compression_level};
```

Comment:
Why do you keep those four values as `int`?

Reply:
ImGui requires it

Reply:
Sorry got this mixed up as well, addressed in my last commit